Source/EasyNetQ/RabbitBus.cs

```diff
@@ -74,20 +91,37 @@ public class RabbitBus : IBus
             return PublishAsync(message, conventions.TopicNamingConvention(typeof(T)));
         }
 
-        public virtual async Task PublishAsync<T>(T message, string topic) where T : class
+        public virtual Task PublishAsync<T>(T message, string topic) where T : class
         {
             Preconditions.CheckNotNull(message, "message");
             Preconditions.CheckNotNull(topic, "topic");
-            var messageType = typeof (T);
+
+            return PublishAsync(message, c => c.WithTopic(topic));
+        }
+
+        public virtual async Task PublishAsync<T>(T message, Action<IPublishConfiguration> configure) where T : class
+        {
+            Preconditions.CheckNotNull(message, "message");
+            Preconditions.CheckNotNull(configure, "configure");
```

Comment:
These checks have small technical flaw: they will throw exception only at `await` call.  So this test is failing

    [Test]
        public void Should_throw()
    {
        var testLocalBus = RabbitHutch.CreateBus("host=localhost;prefetchcount=1");
        Assert.Throws<ArgumentNullException>(() => testLocalBus.PublishAsync(default(string), c => { }));
    }

Old `PublishAsync` method had this behavior too and I haven't changed it.  Should it be fixed in a separate PR or it isn't a problem?